src/keys/k256.rs

```diff
@@ -38,35 +40,28 @@ impl EnrKey for k256_crate::SecretKey {
             .ok_or_else(|| DecoderError::Custom("Unknown signature"))?;
 
         // should be encoded in compressed form, i.e 33 byte raw secp256k1 public key
-        Ok(k256_crate::EncodedPoint::from_bytes(pubkey_bytes)
+        Ok(VerifyKey::new(pubkey_bytes)
             .map_err(|_| DecoderError::Custom("Invalid Secp256k1 Signature"))?)
     }
 }
 
-impl EnrPublicKey for k256_crate::EncodedPoint {
+impl EnrPublicKey for VerifyKey {
     fn verify_v4(&self, msg: &[u8], sig: &[u8]) -> bool {
-        let digest = Keccak256::new().chain(msg);
         if let Ok(sig) = k256_crate::ecdsa::Signature::try_from(sig) {
-            if let Ok(verifier) = Verifier::new(self.as_bytes()) {
-                if verifier.verify_digest(digest, &sig).is_ok() {
-                    return true;
-                }
-            }
+            return self
+                .verify_digest(Keccak256::new().chain(msg), &sig)
+                .is_ok();
         }
         false
     }
 
     fn encode(&self) -> Vec<u8> {
         // serialize in compressed form: 33 bytes
-        self.compress().as_bytes().to_vec()
+        self.to_bytes().to_vec()
     }
 
     fn encode_uncompressed(&self) -> Vec<u8> {
-        k256_crate::AffinePoint::from_encoded_point(self)
-            .unwrap()
-            .to_encoded_point(false)
-            .as_bytes()[1..]
-            .to_vec()
+        EncodedPoint::from(self).decompress().unwrap().to_bytes()[1..].to_vec()
```

Comment:
As of the latest `elliptic-curve` release (v0.6.2), the following should work:

```suggestion
        EncodedPoint::from(self).to_untagged_bytes().unwrap().to_vec()
```